contracts/RealmDiamond/facets/ParcelRolesRegistryFacet.sol

```diff
@@ -30,16 +30,42 @@ contract ParcelRolesRegistryFacet is Modifiers, IERC7432 {
   function grantRole(Role calldata _role) external override onlyValidRole(_role.roleId) onlyRealm(_role.tokenAddress) {
     require(_role.expirationDate > block.timestamp, "ParcelRolesRegistryFacet: expiration date must be in the future");
 
-    // Deposit NFT if necessary and get the original owner
     address _originalOwner = _depositNft(_role.tokenAddress, _role.tokenId);
 
-    // Get role data from storage
     RoleData storage _roleData = s.erc7432_roles[_role.tokenAddress][_role.tokenId][_role.roleId];
 
-    // Check if the role is either revocable or expired
     require(_roleData.revocable || _roleData.expirationDate < block.timestamp, "ParcelRolesRegistryFacet: role must be expired or revocable");
 
-    s.erc7432_roles[_role.tokenAddress][_role.tokenId][_role.roleId] = RoleData(_role.recipient, _role.expirationDate, _role.revocable, _role.data);
+    s.erc7432_roles[_role.tokenAddress][_role.tokenId][_role.roleId] = RoleData(_role.recipient, _role.expirationDate, _role.revocable);
+
+    if (_role.roleId == keccak256("AlchemicaChanneling()") || _role.roleId == keccak256("EmptyReservoir()")) {
+      require(_role.data.length > 0, "ParcelRolesRegistryFacet: No informations provided in ProfitShare");
```

Comment:
Do we need this require? Won't it fail on decoding? It would be good to check that it fails via a test

Reply:
This is covered in this testcase
```
 it("should revert when granting AlchemicaChanneling role without ProfitShare data", async () => {
      const roleWithoutProfitShare = {
          roleId: ROLE_ALCHEMICA_CHANNELING,
          tokenAddress: mockERC721.address,
          tokenId: 1,
          recipient: borrower.address,
          expirationDate: Math.floor(Date.now() / 1000) + ONE_DAY,
          revocable: true,
          data: "0x",
      };
  
      await expect(
          parcelRolesRegistryFacet.connect(owner).grantRole(roleWithoutProfitShare)
      ).to.be.revertedWith("ParcelRolesRegistryFacet: No informations provided in ProfitShare");
  });
```

In my point of view, this require will avoid to call decodeProfitShare if there's nothing inside of data but if in the front end will have this validation, I think we can remove. 

this is more to save the function call.